spark-shell/src/main/scala/org/apache/mahout/sparkbindings/shell/MahoutSparkILoop.scala

```diff
@@ -48,55 +77,63 @@ class MahoutSparkILoop extends SparkILoop {
 
     conf.set("spark.executor.memory", "1g")
 
-    sparkContext = mahoutSparkContext(
+    _interp.sparkContext = mahoutSparkContext(
       masterUrl = master,
       appName = "Mahout Spark Shell",
       customJars = jars,
       sparkConf = conf
     )
 
-    echo("Created spark context..")
+    echoToShell("Created spark context..")
     sparkContext
   }
 
+  // need to change our SparkDistributedContext name to 'sc' since  we cannot override the
+  // private sparkCleanUp() method.
+  // this is technically not part of Sparks explicitly defined Developer API though
+  // nothing in the SparkILoopInit.scala file is marked as such.
   override def initializeSpark() {
-    intp.beQuietDuring {
-      command("""
+    _interp.beQuietDuring {
+      _interp.interpret("""
 
-         @transient implicit val sdc: org.apache.mahout.math.drm.DistributedContext =
+         @transient implicit val sc: org.apache.mahout.math.drm.DistributedContext =
```

Comment:
yes.

Except we are already calling "createMahoutContext" during createSparkContext, so we just need to stash its output as Mahout context private attribute and pass it in here to implicit. (perhaps via a public getter or val).

as for deinitalization, you are right -- all it does i think it stops the spark context.